AutoCollection/diagnostic-channel/tedious.sub.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import TelemetryClient = require("../../Library/TelemetryClient");
+import { channel, IStandardEvent } from "diagnostic-channel";
+
+import { ITediousData } from "diagnostic-channel-publishers/dist/src/tedious.pub";
+
+let clients: TelemetryClient[] = [];
+
+export const subscriber = (event: IStandardEvent<ITediousData>) => {
+    clients.forEach((client) => {
+        const q = event.data.query;
+        const sql = (q.preparable && q.preparable.text) || q.plan || q.text || "unknown query";
+        const success = !event.data.error;
+        const conn = `${event.data.database.host}:${event.data.database.port}`;
+        client.trackDependency({
+            target: conn,
+            data: sql,
+            name: sql,
```

Comment:
Is name same as data on purpose? Is it possible that data is long. Does name has character length limit or something?

Reply:
Yes, the pattern for this repo seems to match the name with the data field for DB RDDs as the SQL command. e.g. the Mongo subscriber

https://github.com/microsoft/ApplicationInsights-node.js/blob/f7d6505e164fd5cfc42c6c0be35441fe76395de2/AutoCollection/diagnostic-channel/mongodb.sub.ts#L17-L22

All the sanitization/truncation for strings are handled at envelope creation. For example, here is where `name` is truncated:
https://github.com/microsoft/ApplicationInsights-node.js/blob/2855952b7348b407f370bef4a10af006425f6508/Library/EnvelopeFactory.ts#L118-L123